OpenRA.Mods.Common/Scripting/Properties/ScaredCatProperties.cs

```diff
@@ -0,0 +1,32 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptPropertyGroup("Movement")]
+	public class ScaredCatProperties : ScriptActorProperties, Requires<ScaredyCatInfo>
+	{
+		public ScaredCatProperties(ScriptContext context, Actor self)
+			: base(context, self)
+		{ }
+
+		[ScriptActorPropertyActivity]
+		[Desc("Makes the unit automatically run around and become faster.")]
+		public void Panic()
+		{
+			Self.Trait<ScaredyCat>().Panic();
```

Comment:
This will crash hard if the actor does not have a `ScaredyCat` trait. Throwing an exception at least provides debugging information.

Reply:
That won't be necessary due to `Requires<ScaredyCatInfo>` but I can cache this trait lookup in the constructor.